src/containers/views/DataFileNew.js

```diff
@@ -78,57 +69,69 @@ export class DataFileNew extends Component {
     }
   };
 
-  renderGUInputs() {
+  renderAside() {
+    const { datafileChanged, fieldChanged, updated } = this.props;
+
     return (
-      <form className="datafile-path">
-        <fieldset className="filename">
-          <legend>Path (without extension)</legend>
-          <input
-            type="text"
-            id="guiPath"
-            onChange={this.handleChange}
-            placeholder="filename"
-          />
-        </fieldset>
-        <fieldset className="file-type">
-          <legend>File Type</legend>
-          <select
-            id="extn"
-            value={this.state.extn}
-            onChange={this.handleChange}
-          >
-            <option value=".yml">YAML</option>
-            <option value=".json">JSON</option>
-          </select>
-        </fieldset>
-      </form>
+      <div className="content-side">
+        <Button
+          onClick={this.handleClickSave}
+          type="create"
+          active={datafileChanged || fieldChanged}
+          triggered={updated}
+          block
+        />
+        <Button
+          onClick={this.toggleView}
+          type="view-toggle"
+          triggered={this.state.guiView}
+          active
+          block
+        />
+      </div>
     );
   }
 
+  renderContentBody() {
+    const { datafileChanged, onDataFileChanged } = this.props;
+
+    if (this.state.guiView) {
+      return (
+        <div className="content-body">
+          <DataGUI
+            source={{ content: {} }}
```

Comment:
We can define `defaultProps` for this in `DataGUI`.

Reply:
Resolved..